core/src/main/java/software/amazon/awssdk/util/SdkRuntime.java

```diff
@@ -20,9 +20,8 @@
 
     /**
      * Returns true if the current operation should abort; false otherwise.
-     * Note the interrupted status of the thread is cleared by this method.
```

Comment:
This can be called at various times during the execution of the wrapped pipeline so the flag gets cleared by the time control returns to `ClientExecutionTimedStage`and it becomes impossible to know to rethrow `ClientExecutionTimeoutException` for the scenario where flag was reset and the IE was not thrown.